edisgo/io/db.py

```diff
@@ -169,9 +175,27 @@ def engine(path: Path | str = None, ssh: bool = False) -> Engine:
     """
 
     if not ssh:
+        # Github Actions KHs token
+        if "TOEP_TOKEN_KH" in os.environ:
+            env_var = os.environ["TOEP_TOKEN_KH"]
+        else:
+            try:
+                with open(token) as file:
+                    env_var = file.read().strip()
+            except FileNotFoundError:
+                if ".txt" in token:
+                    raise FileNotFoundError(f"File {token} not found.")
+                else:
+                    env_var = token
+
+            # Check if the token format is valid
+            if not re.match(r"^[a-f0-9]{40}$", env_var):
+                raise ValueError(
+                    "Invalid token format. Expected a 40-character hexadecimal string."
+                )
         database_url = "toep.iks.cs.ovgu.de"
         return create_engine(
-            "postgresql+oedialect://:@" f"{database_url}",
+            f"postgresql+oedialect://:{env_var}@{database_url}",
```

Comment:
It would be good if it would be still possible to use the engine without the token as unregistered user. This should still be the default.